server/src/main/java/io/deephaven/server/appmode/ApplicationServiceGrpcImpl.java

```diff
@@ -363,12 +367,17 @@ public void close() {
             remove(this);
         }
 
-        // must be sync wrt parent
+        /**
+         * Sends an update to the subscribed client. Returns true if successful - if false, the client is no longer
+         * listening and this subscription should be canceled after iteration.
```

Comment:
Looking at callers of `send` - do we also need to do an onCancel call in `listFields`?

Reply:
Probably, though a) it is a weak ref anyway, and b) the subscription won't be in the list so doesnt need to be removed. I'll make the change in this patch, though it probably should be incorporated into a bigger change that generally cleans up these patterns, rather than fixing each piecemeal as we've been doing.